README.md

```diff
@@ -29,3 +29,126 @@ OpenEBS/project
 ```
 
 > Note: Developer should keep `integration_test` completely decoupled from the rest of the project packages.
+
+## Instantiation
+
+Developer has to instantiate CITF using `NewCITF(configFilePath)` which will initialize the CITF instance with all the required configurations. 
+
+CITF has four components:- 
+- Environment - To Setup or TearDown the platform such as minikube, GKE, AWS etc.
+- K8S - K8S will have Kubernetes ClientSet & Config.
+- Docker - Docker will be used for docker related operations.
+- DebugEnabled - for verbose log.
+
+> Currently CITF supports minikube only.
+
+Developer can pass environment according to their requirements.
+
+By default it will take Minikube as environment.
+
+## Confiuration
+
+To configure the environment of CITF, there are three ways:-
+ - [Environment Variable](#environment-variable)
+ - [Config File](#config-file)
+ - [Default Config](#default-config)
+
+### Environment Variable
+
+At the time of instantiation of CITF, developer can set an environment variable. Environment variable name should be `CITF_CONF_ENVIRONMENT`.
+
+For example:- `export CITF_CONF_ENVIRONMENT = minikube`
```

Comment:
Why we are not keeping it as ENVIRONMENT only instead of CITF_CONF_ENVIRONMENT as we are using environment in config file ?

Reply:
we want to separate CITF related environment variables from other environment variables. That's the only reason.

Reply:
This way environment variables are self explainatory that it is related to CITF configuration.

Reply:
Right ! So we can do it other way around !